StandaloneMmPkg/Drivers/StandaloneMmIplPei/StandaloneMmIplPei.c

```diff
@@ -410,25 +410,36 @@ CreatMmHobList (
   //
   PlatformHobSize = 0;
   Status          = CreateMmPlatformHob (NULL, &PlatformHobSize);
-  if (Status == RETURN_BUFFER_TOO_SMALL) {
+  if (Status == EFI_SUCCESS) {
+    DEBUG ((DEBUG_INFO, "%a: The size of platform MM HOBs to create is 0\n", __func__));
+    return NULL;
+  }
+
+  if (Status != RETURN_BUFFER_TOO_SMALL) {
+    DEBUG ((DEBUG_ERROR, "%a: Unable to determine size of platform MM HOBs to create\n", __func__));
+    ASSERT (Status == RETURN_BUFFER_TOO_SMALL);
+    return NULL;
+  }
+
+  if (PlatformHobSize == 0) {

```

Comment:
Is this checking needed? Shouldn't this already taken care by line 413, PlatformHobSize ==0 when status == EFI_SUCCESS?

Reply:
Got it. Thanks @niruiyu